anomalib/data/ucsd_ped.py

```diff
@@ -0,0 +1,269 @@
+    """
+    folders = [filename for filename in sorted(Path(path).glob("*/*")) if os.path.isdir(filename)]
+    folders = [folder for folder in folders if len(list(folder.glob("*.tif"))) > 0]
+
+    samples_list = [(str(path),) + folder.parts[-2:] for folder in folders]
+    samples = DataFrame(samples_list, columns=["root", "folder", "image_path"])
+
+    samples.loc[samples.folder == "Test", "mask_path"] = samples.image_path.str.split(".").str[0] + "_gt"
+    samples.loc[samples.folder == "Test", "mask_path"] = samples.root + "/" + samples.folder + "/" + samples.mask_path
+    samples.loc[samples.folder == "Train", "mask_path"] = ""
+
+    samples["image_path"] = samples.root + "/" + samples.folder + "/" + samples.image_path
+
+    samples.loc[samples.folder == "Train", "split"] = "train"
+    samples.loc[samples.folder == "Test", "split"] = "test"
+
+    if split:
+        samples = samples[samples.split == split]
+        samples = samples.reset_index(drop=True)
+
+    return samples
+
+
+class UCSDpedClipsIndexer(ClipsIndexer):
+    """Clips class for UCSDped dataset."""
+
+    def get_mask(self, idx) -> Optional[Tensor]:
+        """Retrieve the masks from the file system."""
+
+        video_idx, frames_idx = self.get_clip_location(idx)
+        mask_folder = self.mask_paths[video_idx]
+        if mask_folder == "":  # no gt masks available for this clip
+            return None
+        frames = self.clips[video_idx][frames_idx]
+
+        mask_frames = sorted(glob.glob(mask_folder + "/*"))
+        mask_paths = [mask_frames[idx] for idx in frames.int()]
+
+        masks = torch.stack([Tensor(cv2.imread(mask_path, flags=0)) / 255.0 for mask_path in mask_paths])
+        return masks
+
+    def _compute_frame_pts(self) -> None:
+        """Retrieve the number of frames in each video."""
+        self.video_pts = []
+        for video_path in self.video_paths:
+            n_frames = len(glob.glob(video_path + "/*"))
+            self.video_pts.append(Tensor(range(n_frames)))
+
+        self.video_fps = [None] * len(self.video_paths)  # fps information cannot be inferred from folder structure
+
+    def get_clip(self, idx: int) -> Tuple[torch.Tensor, torch.Tensor, Dict[str, Any], int]:
+        """Gets a subclip from a list of videos.
+
+        Args:
+            idx (int): index of the subclip. Must be between 0 and num_clips().
+
+        Returns:
+            video (Tensor)
+            audio (Tensor)
+            info (Dict)
+            video_idx (int): index of the video in `video_paths`
+        """
+        if idx >= self.num_clips():
+            raise IndexError(f"Index {idx} out of range ({self.num_clips()} number of clips)")
+        video_idx, clip_idx = self.get_clip_location(idx)
+        video_path = self.video_paths[video_idx]
+        clip_pts = self.clips[video_idx][clip_idx]
+
+        frames = sorted(glob.glob(video_path + "/*"))
```

Comment:
pathlib

Reply:
This has to be glob as well, since `video_path` is a string. Otherwise it would become `len(list(Path(video_path).glob("*")))` which I find a bit verbose.